backend/ics.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/samouraiworld/zenao/backend/zeni"
+	"go.uber.org/zap"
+)
+
+const icsDateTime = "20060102T150405Z"
+
+// see: https://datatracker.ietf.org/doc/html/rfc5545
+func GenerateICS(event *zeni.Event, zenaoEmail string, logger *zap.Logger) []byte {
+	uid := fmt.Sprintf("evt_%s@zenao.io", event.ID)
+	summary := formatICSText(event.Title)
+	dtstamp := time.Now().UTC().Format(icsDateTime)
+	dtstart := event.StartDate.Format(icsDateTime)
+	dtend := event.EndDate.Format(icsDateTime)
+	eventURL := fmt.Sprintf("https://zenao.io/event/%s", event.ID)
+	description := formatICSText(fmt.Sprintf("You are invited to %s!", event.Title))
+	location, err := zeni.LocationToString(event.Location)
+	if err != nil {
+		logger.Error("failed to convert location to string", zap.Error(err), zap.String("event-id", event.ID))
+		location = ""
+	}
+	location = formatICSText(location)
+	organizerFormatted := fmt.Sprintf("CN=Zenao:mailto:%s", zenaoEmail)
+
+	// XXX: save event edit history in the database to track sequence number
+	ics := fmt.Sprintf(`BEGIN:VCALENDAR
```

Comment:
maybe use a serializer instead of custom template? https://github.com/arran4/golang-ical